### DISCUSSION TITLE
Add more actionable elements to the site homepage

### DISCUSSION THREAD

**Discussion by visortelle** [apache/pulsar#22189] [Ideas]
- Call to star Pulsar on GitHub ⭐️

We can use it as a default announcement bar when there are no upcoming events or other announcements.

**Apache Doris**
<img width="1440" alt="Screenshot 2024-03-04 at 3 27 58 PM" src="https://github.com/apache/pulsar/assets/9302460/c23ca817-2d52-4788-ab2d-07e23756b8c7">

**Apache Pinot**
<img width="1440" alt="Screenshot 2024-03-04 at 3 28 08 PM" src="https://github.com/apache/pulsar/assets/9302460/80e1ca51-c950-42dc-8590-88c973ce2fd0">

---

- Call to join Slack

**Apache Doris**

<img width="1063" alt="Screenshot 2024-03-04 at 3 30 36 PM" src="https://github.com/apache/pulsar/assets/9302460/caf2ca92-7208-451c-adff-d563706a80d6">

**Apache Pinot**

<img width="1018" alt="Screenshot 2024-03-04 at 3 31 50 PM" src="https://github.com/apache/pulsar/assets/9302460/8bd159a4-5591-4d01-813b-9d5cf0311ca3">

---

- Add a few lines of code that show how to launch Pulsar

In the case of Pulsar, it may be the follwing one-liner:

```sh
docker run -it -p 6650:6650 -p 8080:8080 apachepulsar/pulsar:3.2.0 bin/pulsar standalone
```

**Apache Pinot**
<img width="1344" alt="Screenshot 2024-03-04 at 3 33 34 PM" src="https://github.com/apache/pulsar/assets/9302460/e58b2584-6cb2-4f91-b553-e1d5dd076ce0">

**Comment by jak78** [comment]
+100

**Comment by visortelle** [comment]
@asafm you asked not to merge any site PRs without approval by a reviewer.

Probably, this proposal should also be approved by someone to avoid the extra work. But I have no idea by whom 🤷‍♂️

Let's say @jak78 or @lhotari gave a 👍 for a proposal. Should I start the implementation? There is no big green "Approve" button in GitHub discussions.

This page states that I should first discuss the change with the committer https://pulsar.apache.org/contribute/
In this case https://github.com/apache/pulsar-site/pull/809#issuecomment-1976880355 I discussed it with @dave2wave who is the PMC member.

Maybe I should go with the dev mailing list, or even with the PIP process?

The mailing list is not good for discussions about the site improvements. Without a convenient way to attach images or videos, it's mostly useless.

Probably I should cross-link the mailing list thread and the GitHub discussion.

But this way we have multiple places where people discuss the same thing and cross-link pages multiple times. I saw that ASF states that mailing lists are good for long-living projects because services like GitHub are born and die, but email is eternal and useful when you need to find something that happened 15 years ago.

Maybe it makes sense to just write a tool that periodically dumps GitHub issues and discussions to the mailing list archive format to be sure they are stored forever?

Another problem with the dev mailing list is that only developers see it. I would like to see comments and reactions from users too. Users are part of the community! The users@pulsar.apache.org mailing list is rather dead than alive, everyone uses GitHub.

I don't think it should be the PIP process for such small site changes. It would take too much time and community resources. For the site, it makes sense to quickly check if some improvement works or not and probably revert the change or find an alternative.

If you read this to the end, then thank you for that. The comment turned out to be more than I expected 😆

**Comment by dave2wave** [comment]
> On Mar 4, 2024, at 1:56 PM, Kiryl Valkovich 🛸 ***@***.***> wrote:
> 
> 
> @asafm <https://github.com/asafm> you asked not to merge any site PRs without approval by a reviewer.
> 
> Probably, this proposal should also be approved by someone to avoid the extra work. But I have no idea by whom 🤷‍♂️
> 
> Let's say @jak78 <https://github.com/jak78> or @lhotari <https://github.com/lhotari> gave a 👍 for a proposal. Should I start the implementation? There is no big green "Approve" button in GitHub discussions.
> 
> This page states that I should discuss the change with the committer https://pulsar.apache.org/contribute/ <https://pulsar.apache.org/contribute/>
> In this case apache/pulsar-site#809 (comment) <https://github.com/apache/pulsar-site/pull/809#issuecomment-1976880355> I discussed it with @dave2wave <https://github.com/dave2wave> who is the PMC member.
> 
> Maybe I should go with the dev mailing list, or even with the PIP process?
> 
> 

You should discuss open the dev mailing list. And I suggest that you start with the subject “[DISCUSS] ….”. I don’t think a PIP is needed. The dev list preserves the discussion on ASF resources forever.
> The mailing list is not good for discussions about the site improvements. Without a convenient way to attach images or videos, it's mostly useless.
> 
> Probably I should cross-link the mailing list thread and the GitHub discussion.
> 
> 
Definitely link to the GitHub discussion in the email.
> But this way we have multiple places where people discuss the same thing and cross-link pages multiple times. I saw that ASF states that mailing lists are good for long-living projects because services like GitHub are born and die, but email is eternal and useful when you need to find something that happened 15 years ago.
> 
> Maybe it makes sense to just write a tool that periodically dumps GitHub issues and discussions to the mailing list archive format to be sure they are stored forever?
> 
> 

We have the ability to send GitHub events to a mailing list. Ask on the dev@ mailing list to start that discussion.
> Another problem with the dev mailing list is that only developers see it. I would like to see comments and reactions from users too. Users are part of the community! The ***@***.*** ***@***.***> mailing list is rather dead than alive, everyone uses GitHub.
> 
> 

Anyone can join the dev2 mailing list and at lists.apache.org <http://lists.apache.org/> you will find email from 25 years ago. Source control was typically CVS back then.
> I don't think it should be the PIP process for such small site changes. It would take too much time and community resources. For the site, it makes sense to quickly check if some improvement works or not and probably revert the change or find an alternative.]
> 
> 

I agree. No PIP.

**Reply by visortelle** [reply]
@dave2wave thank you for the clarification. I sent the proposal to the mailing list.

**Comment by asafm** [comment]
The bar on top:
I think that people notice the bar when it's new :) If it is always there, they ignore it. 

I liked the idea of adding a link to GitHub and Slack. I prefer adding them here:
<img width="559" alt="image" src="https://github.com/apache/pulsar/assets/989425/84e73e98-03e1-4453-a0d2-41eaee7cda7d">

I'm ok with showing how easy it is to spin up Pulsar locally. What were your thoughts on the structure of that section (title, paragraph, etc)?

**Reply by visortelle** [reply]
> What were your thoughts on the structure of that section (title, paragraph, etc)?

---
Run Pulsar in a single command:

`docker run ... bin/pulsar standalone`

Observe the other deployment options

| Helm | Bare metal | Docker | Docker-compose |

---

**Reply by visortelle** [reply]
> I think that people notice the bar when it's new :) If it is always there, they ignore it.

We can use two very different colors or icons for that.

**Reply by asafm** [reply]
For the title "Run Pulsar in a single command:" how about:

"Deploy easily".
or
"Try it out"

Regarding the upper bar with GitHub/Slack link: Sorry, I'm not a fan of the idea. Let's see what others think.

**Reply by jak78** [reply]
I prefer 'Try it out' to 'Deploy easily' because using Pulsar standalone in production deployments is not recommended.
